src/contexts/home/HomeContext.tsx

```diff
@@ -0,0 +1,48 @@
+import React, {
+  createContext,
+  useState,
+  Props,
+  useMemo,
+  useCallback,
+} from "react";
+
+import GithubService from "~/services/GithubService";
+import { Repository } from "~/utils/model/Repository";
+
+interface HomeContextValue {
+  username: string;
+  repos: Repository[];
+  totalStars: number;
+  searchRepo: (name: string) => void;
+}
+
+export const HomeContext = createContext({} as HomeContextValue);
+
+export const HomeProvider = (props: Props<HTMLElement>) => {
+  const { children } = props;
+
+  const [username, setUsername] = useState("");
+  const [repos, setRepos] = useState([] as Repository[]);
```

Comment:
이렇게 제네릭 형식으로 선언하는건 어떨까요??
```suggestion
  const [repos, setRepos] = useState<Repository[]>([]);
```

Reply:
as를 뒤늦게 처리해서 놓치고 있었네요 ㅋㅋ

Reply:
오 제네릭으로 처리할 수 도 있군요. 처음 알았습니다 :+1: